src/pages/register-a-beacon/application-complete.tsx

```diff
@@ -89,15 +90,20 @@ const ApplicationCompleteYourBeaconRegistryAccount: FunctionComponent =
 export const getServerSideProps: GetServerSideProps = withSession(
   withContainer(async (context: BeaconsGetServerSidePropsContext) => {
     /* Retrieve injected use case(s) */
-    const { submitRegistration, getAccountHolderId } = context.container;
+    const { getDraftRegistration, submitRegistration, getAccountHolderId } =
+      context.container;
 
     /* Page logic */
     if (!verifyFormSubmissionCookieIsSet(context))
       return redirectUserTo(PageURLs.start);
 
     try {
+      const draftRegistration: DraftRegistration = await getDraftRegistration(
+        context.req.cookies[formSubmissionCookieId]
+      );
+
       const result = await submitRegistration(
-        context.req.cookies[formSubmissionCookieId],
+        draftRegistration,
         await getAccountHolderId(context.session)
```

Comment:
Separate out concerns of "Get the user's DraftRegistration from cache" and "Submit DraftRegistration".  Needed to do this for the Cypress test, but I think it's also a bit cleaner.